File: app/src/main/java/com/example/instagram/modal/Photo.java
package com.example.instagram.modal;

import android.util.Log;
import android.widget.ImageView;

import androidx.databinding.BindingAdapter;

import com.bumptech.glide.Glide;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Photo {

    private final String MediaType = "image";

    @SerializedName("id")
    @Expose
    private Integer id;
    @SerializedName("url")
    @Expose
    private String url;
    @SerializedName("alt")
    @Expose
    private String alt;
    @SerializedName("photographer")
    @Expose
    private String photographer;

    @SerializedName("src")
    @Expose
    private Src src;

    public Src getSrc() {
        return src;
    }

    public void setSrc(Src src) {
        this.src = src;
    }

    @BindingAdapter("glide_image")
    public static void loadImage(ImageView imageView, String url){
        if (url != null && !url.isEmpty()){
            Glide.with(imageView.getContext()).load(url).into(imageView);
        }else {
            Log.d("GlideLoad", "Failed");
        }

    }

    public String getAlt() {
        return alt;
    }

    public void setAlt(String alt) {
        this.alt = alt;
    }

    public String getPhotographer() {
        return photographer;
    }

    public void setPhotographer(String photographer) {
        this.photographer = photographer;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

}

File: app/src/main/java/com/example/ktinsta/modal/Photo.kt
package com.example.ktinsta.modal

import android.util.Log
import android.widget.ImageView
import androidx.databinding.BindingAdapter
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.example.instagram.R
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class Photo (private val type : String = ""){


    @SerializedName("id")
    @Expose
    var id: Int? = null

    @SerializedName("url")
    @Expose
    var url: String? = null

    @SerializedName("alt")
    @Expose
    var alt: String? = null

    @SerializedName("photographer")
    @Expose
    var photographer: String? = null

    @SerializedName("src")
    @Expose
    var src: Src? = null


    companion object {
        @JvmStatic
        @BindingAdapter("glide_image")
        fun loadImage(imageView: ImageView, url: String?) {
            Glide.with(imageView.context)
                .load(url)
                .diskCacheStrategy(DiskCacheStrategy.AUTOMATIC)
                .error(R.drawable.connection_error)
                .into(imageView)

        }
    }
}
